tools/build_script_generator/cmake/resources/CMakeLists.txt.in

```diff
@@ -0,0 +1,52 @@
+# This file was autogenerated by the modm cmake builder.
+# Do not modify!
+
+CMAKE_MINIMUM_REQUIRED(VERSION 3.0)
+PROJECT({{ project_name }})
+
+FILE(GLOB_RECURSE APP_STARTUP *.sx)
+SET_SOURCE_FILES_PROPERTIES(${APP_STARTUP} PROPERTIES LANGUAGE CXX)
+FILE(GLOB APP_SOURCE_FILES *.c *.cpp *.hpp *.h)
+
+SET(SOURCE_FILES
+    ${APP_STARTUP}
+%% for source_file in files
+    {{ source_file }}
+%% endfor
```

Comment:
Won't this duplicate the `.sx` files in this list? `APP_STARTUP` finds them via CMake, and the `files` list contains it again.

Reply:
It does, but this does not triggers any problems because CMake seems filters out duplicates. I've reworked code a bit to avoid duplication in the first place. This part is a little bit tricky because CMake does not recognize .sx as a valid extension associated with any language. I ought to workaround this by setting file properties explicitly.